contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+            } else {
+                revert NotEnoughWethForSwap(balance, _amountToSwap);
+            }
+        }
+
+        // emit an event so it is easier to find correct values off-chain
+        (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionBeforeSwap(WETHPositionBalance, OETHbPositionBalance);
+
+        // Swap it
+        uint256 amountReceived = swapRouter.exactInputSingle(
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: _swapWETH ? WETH : OETHb,
+                tokenOut: _swapWETH ? OETHb : WETH,
+                tickSpacing: tickSpacing, // set to 1
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: _amountToSwap,
+                amountOutMinimum: _minTokenReceived, // slippage check
+                // just a rough sanity check that we are within 0 -> 1 tick
+                // a more fine check is performed in _checkLiquidityWithinExpectedShare
+                sqrtPriceLimitX96: _swapWETH ? sqrtRatioX96Tick0 : sqrtRatioX96Tick1
+            })
+        );
+
+        // emit an event so it is easier to find correct values off-chain
+        (WETHPositionBalance, OETHbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionAfterSwap(WETHPositionBalance, OETHbPositionBalance);
+    }
+
+    /**
+     * @dev Perform a swap so that after the swap the ticker has the desired WETH to OETHb token share.
+     */
+    function _addLiquidity() internal {
+        uint256 WETHBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 OETHbBalance = IERC20(OETHb).balanceOf(address(this));
+        require(WETHBalance > 0, "Must add some WETH");
+        uint256 OETHbRequired;
+
+        if (tokenId == 0) {
+            // supply token amounts according to target poolWethShare amount
+            OETHbRequired = WETHBalance.divPrecisely(poolWethShare)
+                .mulTruncate(1e18 - poolWethShare);
+
+        } else {
+            // supply the tokens to the pool according to the current position in the pool
+            (uint256 WETHPositionBalance, uint256 OETHbPositionBalance) = getPositionPrincipal();
+
+            require(OETHbPositionBalance > 0, "Can not calculate OETHb required");
+            OETHbRequired = OETHbPositionBalance.divPrecisely(WETHPositionBalance).mulTruncate(WETHBalance);
+        }
+
+        if (OETHbRequired > OETHbBalance) {
+            IVault(vaultAddress).mintForStrategy(OETHbRequired - OETHbBalance);
+        }
+
+        if (tokenId == 0) {
+            uint160 sqrtRatioX96 = getPoolX96Price();
+
+            (uint256 mintedTokenId, uint128 liquidity, uint256 WETHAmountSupplied, uint256 OETHbAmountSupplied) = positionManager.mint(
+                INonfungiblePositionManager.MintParams({
+                    token0: WETH,
+                    token1: OETHb,
+                    tickSpacing: 1,
```

Comment:
Should this use the global `tickSpacing` immutable instead of hardcoded value?

Reply:
nice catch: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/03d9d2c1655412da81ec01fb718454b5c6643079